operatorcourier/api.py

```diff
@@ -106,16 +83,15 @@ def build_verify_and_push(namespace, repository, revision, token,
     :raises OpCourierQuayErrorResponse: When Quay responds with an error
     :raises OpCourierQuayError: When the request fails in an unexpected way
     """
-
-    bundle = build_and_verify(source_dir, yamls, repository=repository,
-                              validation_output=validation_output, output=output)
-
-    with TemporaryDirectory() as temp_dir:
-        with open(os.path.join(temp_dir, 'bundle.yaml'), 'w') as outfile:
-            yaml.dump(bundle, outfile, default_flow_style=False)
-            outfile.flush()
-
-        PushCmd().push(temp_dir, namespace, repository, revision, token)
+    verification = build_and_verify(source_dir, yamls, repository=repository,
+                                    validation_output=validation_output, output=output)
+    if not verification.nested:
+        with TemporaryDirectory() as temp_dir:
+            with open(os.path.join(temp_dir, 'bundle.yaml'), 'w') as outfile:
+                yaml.dump(verification.bundle, outfile, default_flow_style=False)
+            PushCmd().push(temp_dir, namespace, repository, revision, token)
+    else:
+        PushCmd().push(source_dir, namespace, repository, revision, token)
```

Comment:
We should be careful here. My recommendation would be to copy the files into a temp directory and then create the tarball from there. It seems dangerous to use the source files directly as input considering we are making os calls other than read.

Reply:
I'm also curious how this will work when yamls are defined instead of source_dir. (I don't need yamls param for my use cases, so it can be dropped if nobody else is using that)

Reply:
Got it, yamls are supported only as flat